vsphere/README.md

```diff
@@ -1,32 +1,93 @@
+# Configuration
 
-## Configuration
+In the Administration section of vCenter, add a read-only user called datadog-readonly.
 
-Edit the `vsphere.yaml` file to point to your server and port, add your username and password and restart the agent
+Then, create a file `vsphere.yaml` in the Datadog Agent's `conf.d` directory:
 
-## Validation
+```
+init_config:
 
-When you run `datadog-agent info` you should see something like the following:
+instances:
+  - name: main-vcenter # how metrics will be tagged, i.e. 'vcenter_server:main-vcenter'
+    host: <VCENTER_HOSTNAME>          # e.g. myvcenter.example.com
+    username: <USER_YOU_JUST_CREATED> # e.g. datadog-readonly@vsphere.local
+    password: <PASSWORD>
+```
 
-    Checks
-    ======
+Restart the Agent to start sending vSphere metrics and events to Datadog.
 
-        vsphere
-        -----------
-          - instance #0 [OK]
-          - Collected 39 metrics, 0 events & 7 service checks
+## Configuration Options
 
-## Compatibility
+* `ssl_verify` (Optional) - Set to false to disable SSL verification, when connecting to vCenter optional
+* `ssl_capath` (Optional) - Set to the absolute file path of a directory containing CA certificates in PEM format
+* `host_include_only_regex` (Optional) - Use a regex like this if you want only the check to fetch metrics for these ESXi hosts and the VMs running on it
+* `vm_include_only_regex` (Optional) - Use a regex to include only the VMs that are matching this pattern.
+* `include_only_marked` (Optional) - Set to true if you'd like to only collect metrics on vSphere VMs which are marked by a custom field with the value 'DatadogMonitored'. To set this custom field with PowerCLI, use the follow command: `Get-VM <MyVMName> | Set-CustomField -Name "DatadogMonitored" -Value "DatadogMonitored"`
+* `all_metrics` (Optional) - When set to true, this will collect EVERY metric from vCenter, which means a LOT of metrics you probably do not care about. We have selected a set of metrics that are interesting to monitor for you if false.
+* `event_config` (Optional) - Event config is a dictionary. For now the only switch you can flip is collect_vcenter_alarms which will send as events the alarms set in vCenter.
 
-The vsphere check is compatible with all major platforms
+# Validation
+
+Run the Agent's `info` subcommand and look for `vsphere` under the Checks section:
+
+```
+  Checks
+  ======
+    [...]
+
+    vsphere
+    -------
+      - instance #0 [OK]
+      - Collected 26 metrics, 0 events & 1 service check
+
+    [...]
+```
+
+# Compatibility
+
+The vsphere check is compatible with all Windows platforms.
+
+# Metrics
+
+See [metadata.csv](https://github.com/DataDog/integrations-core/blob/master/vsphere/metadata.csv) for a list of metrics provided by this check.
+
+# Events
+
+This check watches vCenter's Event Manager for events and emits them to Datadog. It does NOT emit the following event types:
+
+AlarmStatusChangedEvent:Gray
```

Comment:
No, it comes straight from the code: https://github.com/DataDog/integrations-core/blob/master/vsphere/check.py#L54